components/ControllableExpertiseSelector.js

```diff
@@ -0,0 +1,299 @@
+/* globals $, promptError: false */
+import { useState, useEffect } from 'react'
+import Icon from './Icon'
+import IconButton from './IconButton'
+import styles from '../styles/components/ControllableExpertiseSelector.module.scss'
+import { stringToColor } from '../lib/utils'
+
+export default function ControllableExpertiseSelector() {
+  const [keyphrases, setKeyphrases] = useState({ active: [], inactive: [] })
+  const [includedKeyphrases, setIncludedKeyphrases] = useState([])
+  const [clickedKeyphrase, setClickedKeyphrase] = useState('')
+  const [recommendedPapers, setRecommendedPapers] = useState([])
+  const [activePapers, setActivePapers] = useState({})
+  const [keyphraseColors, setKeyphraseColors] = useState({})
+  const [paperWeights, setPaperWeights] = useState({})
+
+  useEffect(() => {
+    async function fetchKeyphrases() {
+      try {
+        const keyphraseResponse = await fetch(
+          'https://retrievalapp-zso5o2q47q-uc.a.run.app/get_recommendations/mccallum'
+        )
+        const jsonResponse = await keyphraseResponse.json()
+        setKeyphrases({
+          active: Object.keys(jsonResponse.profilekp2selected),
+          inactive: [],
+        })
+        // Compute colors after keyphrases are set
+        // keyphrases state var might not be updated immediately
+        setKeyphraseColors(
+          Object.keys(jsonResponse.profilekp2selected).reduce((acc, kp) => {
+            acc[kp] = stringToColor(kp)
+            return acc
+          }, {})
+        )
+
+        setIncludedKeyphrases(
+          Object.keys(jsonResponse.profilekp2selected).filter(
+            (kp) => jsonResponse.profilekp2selected[kp]
+          )
+        )
+        setRecommendedPapers(jsonResponse.recommendations)
+        setActivePapers(jsonResponse.profilekp2user_papers)
+      } catch (error) {
+        promptError(error.message)
+      }
+    }
+
+    fetchKeyphrases()
+  }, [])
+
+  useEffect(() => {
+    function buildPaperWeightDict(jsonResponse) {
+      const paperWeightDict = {}
+      Object.keys(jsonResponse.profilekp2user_papers).forEach((kp) => {
+        jsonResponse.profilekp2user_papers[kp].papers.forEach((paper, index) => {
+          if (Object.prototype.hasOwnProperty.call(paperWeightDict, paper.title)) {
+            paperWeightDict[paper.title] = {
+              ...paperWeightDict[paper.title],
+              [kp]: jsonResponse.profilekp2user_papers[kp].kp2paper_wts[index],
+            }
+          } else {
+            paperWeightDict[paper.title] = {
+              [kp]: jsonResponse.profilekp2user_papers[kp].kp2paper_wts[index],
+            }
+          }
+        })
+      })
+
+      return paperWeightDict
+    }
+    async function fetchRecommendations() {
+      const baseUrl =
+        'https://retrievalapp-zso5o2q47q-uc.a.run.app/get_recommendations/mccallum'
```

Comment:
can this be a parameter of the component?

does it work for all the venues or only specific ones?

Reply:
Actually this will not work for all venues. This backend caches reviewer data and reviewers will be specific to a venue. Venues may also choose to show "Potential Paper Assignments" using a domain-specific collection e.g. while NLP/ML venues may use papers from those veneues (as we are now) Computer Vision venues may use a different collection of papers.

This can be changed in future designs but for a first run, this is easiest to set up.

Reply:
How do you handle authentication here? is this link public for any user?

Reply:
@carlosmondra indicates that this server must support cross domain so it can be called from openreview.net.

Reply:
> How do you handle authentication here? is this link public for any user?

We havent added any authentication to the API call yet. This is one of our immediate todos - would you prefer for us to complete this before this PR is accepted?

> this server must support cross domain so it can be called from openreview.net.

What do you mean? It should be possible to call the server from openreview.net.

Reply:
Authentication must be supported if we are going to ask the OpenReview users to send and receive sensitive data from your servers.

In order to call to a server with a different domain: from openreview.net to your sheshera.server.com, your server must accept calls from a different domain for security reasons. More info here: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS